modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java

```diff
@@ -422,27 +440,20 @@ public void processOpts() {
 
         if (!interfaceOnly) {
             if (SPRING_BOOT.equals(library)) {
+                if (useSwaggerUI && selectedDocumentationProviderRequiresSwaggerUiBootstrap()) {
+                    supportingFiles.add(new SupportingFile("swagger-ui.mustache", "src/main/resources/static", "swagger-ui.html"));
+                }
+                // rename template to SpringBootApplication.mustache
                 supportingFiles.add(new SupportingFile("openapi2SpringBoot.mustache",
                         (sourceFolder + File.separator + basePackage).replace(".", java.io.File.separator),
-                        "OpenAPI2SpringBoot.java"));
+                        "OpenApiGeneratorApplication.java"));
+                supportingFiles.add(new SupportingFile("SpringBootTest.mustache",
```

Comment:
Should this check if test output is enabled first?

Reply:
Not sure. It is a supportingFile. I know generateModelTests, and generatApilTests but supportingfiles?

Reply:
Good point - it doesn't really fit either of those two categories.

Still feels like it should likely be behind some kind of option since no other tests get generated without tests enabled - thoughts?

Reply:
Hmm. Users who want to create a new SpringBoot project do want a Test. Users who use interfaceOnly / apiFirst probably skip supporting files all together.

Reply:
I think we should implement a global option `generateSupportTests`. But this is way out if scope for this PR.